tests/basic/legacy_shared_sharedv4_service_migration_tests.go

```diff
@@ -0,0 +1,655 @@
+package tests
+
+import (
+	"fmt"
+	"time"
+	"math/rand"
+
+	"github.com/portworx/torpedo/pkg/log"
+	"github.com/portworx/torpedo/pkg/testrailuttils"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/torpedo/drivers/node"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	"github.com/portworx/torpedo/drivers/volume"
+	"k8s.io/apimachinery/pkg/types"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/portworx/torpedo/tests"
+)
+
+const (
+	ubuntu = "ubuntu-app"
+)
+
+// Legacy Shared Volume Create
+// Automatically it should get created as Sharedv4 service volume.
+
+var _ = Describe("{LegacySharedVolumeCreate}", func() {
+	var testrailID = 296369
+	// https://portworx.testrail.net/index.php?/cases/view/296369
+
+	StartTorpedoTest("LegacySharedVolumeAppCreateVolume", "Legacy Shared to Sharedv4 Service CreateVolume", nil, testrailID)
+
+	volumeName := "legacy-shared-volume"
+	stepLog := "Create legacy shared volume and check it got created as sharedv4 service volume"
+	It(stepLog, func() {
+		pxNodes, err := GetStorageNodes()
+		log.FailOnError(err, "Unable to get the storage nodes")
+		pxNode := GetRandomNode(pxNodes)
+		log.Infof("Creating legacy shared volume: %s", volumeName)
+		pxctlCmdFull := fmt.Sprintf("v c --shared=true %s", volumeName)
+		output, err := Inst().V.GetPxctlCmdOutput(pxNode, pxctlCmdFull)
+		log.FailOnError(err, fmt.Sprintf("error creating legacy shared volume %s", volumeName))
+		log.Infof(output)
+		vol, err := Inst().V.InspectVolume(volumeName)
+		log.FailOnError(err, fmt.Sprintf("Inspect volume failed on volume {%v}", volumeName))
+		Expect(vol.Spec.Sharedv4).To(BeTrue(), "sharedv4 volume was not created")
+		Expect(vol.Spec.Shared).To(BeFalse(), "shared volume was created unexpectedly")
+		pxctlCmdFull = fmt.Sprintf("v d %s", volumeName)
+		output, err = Inst().V.GetPxctlCmdOutput(pxNode, pxctlCmdFull)
+		log.FailOnError(err, fmt.Sprintf("error deleting legacy shared volume %s", volumeName))
+	})
+	EndTorpedoTest()
+})
+
+func setCreateLegacySharedAsSharedv4Service(on bool) {
+	pxNodes, err := GetStorageNodes()
+	log.FailOnError(err, "Unable to get storage nodes")
+	pxNode := GetRandomNode(pxNodes)
+	log.Infof("Setting Creation of Legacy shared volumes")
+	var pxctlCmdFull string
+	if on {
+		pxctlCmdFull = fmt.Sprintf("cluster update option --create-legacy-shared-as-sharedv4-service=true")
```

Comment:
you write as `pxctlCmdFull = fmt.Sprintf("cluster update option --create-legacy-shared-as-sharedv4-service=%t",on) `
without using if else